tests/hint.test

```diff
@@ -1,9 +1,7 @@
 ---------------------------------------------------------------------
 RUN tests/newtype-derive.hs --with= --hint=data/hlint.yaml
 FILE tests/newtype-derive.hs
-{-# LANGUAGE DeriveFoldable #-}
-{-# LANGUAGE DeriveFunctor #-}
-{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE DeriveTraversable #-} -- Implies DeriveFoldable and DeriveFunctor
```

Comment:
I don't get it. In this test you remove `DeriveFunctor`, but at the same time add in a hint that DeriveFunctor is not required. I don't get why you have both patches in one code, or how that still passes the test. I suspect I'm missing something.

Reply:
I made two changes:
  - `newtype-deriv.hs`
    - Don't import `DeriveFoldable` or `DeriveFunctor`
    -  If we import them, we'll get new output hinting that they are not required
  - `restricted-extension`
    - Do import `DeriveFoldable` and `DeriveFunctor` (no change)
    - Add the extra warnings now generated to the expected output
    - Intent is to allow this as an incidental test of this PR.

Reply:
OK, that makes sense - very good idea